### PR TITLE
[vk-bootstrap] Update to v0.7

### LINKED ISSUE #29611: [vk-bootstrap] Build error
**Host Environment**

- OS:Windows 10 22H2 19045.2546
- Compiler: MSVC 19.34.31937.0

**To Reproduce**

`vcpkg install vk-bootstrap`

**Failure logs**

```
PS D:\Project\AirEngine> vcpkg install vk-bootstrap
Computing installation plan...
The following packages will be built and installed:
    vk-bootstrap[core]:x86-windows -> 0.5#1
Detecting compiler hash for triplet x86-windows...
Restored 0 package(s) from C:\Users\Air\AppData\Local\vcpkg\archives in 141.2 us. Use --debug to see more details.
Installing 1/1 vk-bootstrap:x86-windows...
Building vk-bootstrap[core]:x86-windows...
-- Note: vk-bootstrap only supports static library linkage. Building static library.
-- Using cached charles-lunarg-vk-bootstrap-142986cdb767037118b687387b097ff6b3e7fe7d.tar.gz.
-- Cleaning sources at D:/Project/vcpkg/buildtrees/vk-bootstrap/src/f6b3e7fe7d-0ef13f4336.clean. Use --editable to skip cleaning for the packages you specify.
-- Extracting source D:/Project/vcpkg/downloads/charles-lunarg-vk-bootstrap-142986cdb767037118b687387b097ff6b3e7fe7d.tar.gz
-- Using source at D:/Project/vcpkg/buildtrees/vk-bootstrap/src/f6b3e7fe7d-0ef13f4336.clean
-- Found external ninja('1.11.0').
-- Configuring x86-windows
-- Building x86-windows-dbg
CMake Error at scripts/cmake/vcpkg_execute_build_process.cmake:134 (message):
    Command failed: E:/Software/CMake/bin/cmake.exe --build . --config Debug --target install -- -v -j17
    Working Directory: D:/Project/vcpkg/buildtrees/vk-bootstrap/x86-windows-dbg
    See logs for more information:
      D:\Project\vcpkg\buildtrees\vk-bootstrap\install-x86-windows-dbg-out.log

Call Stack (most recent call first):
  installed/x64-windows/share/vcpkg-cmake/vcpkg_cmake_build.cmake:74 (vcpkg_execute_build_process)
  installed/x64-windows/share/vcpkg-cmake/vcpkg_cmake_install.cmake:16 (vcpkg_cmake_build)
  ports/vk-bootstrap/portfile.cmake:18 (vcpkg_cmake_install)
  scripts/ports.cmake:147 (include)


error: building vk-bootstrap:x86-windows failed with: BUILD_FAILED
error: Please ensure you're using the latest port files with `git pull` and `vcpkg update`.
Then check for known issues at:
    https://github.com/microsoft/vcpkg/issues?q=is%3Aissue+is%3Aopen+in%3Atitle+vk-bootstrap
You can submit a new issue at:
    https://github.com/microsoft/vcpkg/issues/new?template=report-package-build-failure.md&title=[vk-bootstrap]+Build+error
Include '[vk-bootstrap] Build error' in your bug report title, the following version information in your bug description, and attach any relevant failure logs from above.
    vcpkg-tool version: 2023-01-24-8a88d63f241d391772fbde69af9cab96c3c64c75
    vcpkg-scripts version: 36fb23307 2023-02-10 (2 days ago)

```
[install-x86-windows-dbg-out.log](https://github.com/microsoft/vcpkg/files/10715688/install-x86-windows-dbg-out.log)

**Additional context**

### PR DESCRIPTION
**Pull request by freestriker**
The existing version is too old, and when a higher version of vulkan sdk is installed locally, it causes inconsistencies in the header files of vk-bootstrap and vulkan sdk, resulting in errors such as #29611

Fixes #29611 

- [x] Changes comply with the [maintainer guide](https://github.com/microsoft/vcpkg-docs/blob/main/vcpkg/contributing/maintainer-guide.md)
- [x] SHA512s are updated for each updated download
- [x] The "supports" clause reflects platforms that may be fixed by this new version
- [x] Any fixed [CI baseline](https://github.com/microsoft/vcpkg/blob/master/scripts/ci.baseline.txt) entries are removed from that file.
- [x] Any patches that are no longer applied are deleted from the port's directory.
- [ ] The version database is fixed by rerunning `./vcpkg x-add-version --all` and committing the result.
- [x] Only one version is added to each modified port's versions file.

### CODE REVIEW

**Comment by FrankXie05** [comment]
Error: 
Your local commit contains the wrong hash.
```
0.5#1 is declared with 5b4e562582567c73a3bad2120a3b23e0b6b32dde, but the local port has a different SHA 220863389d6794148e46747023364d9f6c17c2e8.
Please update the port's version fields and then run:
vcpkg x-add-version vk-bootstrap
```

File: versions/baseline.json [hunk 1]
```diff
@@ -8353,8 +8353,8 @@
       "port-version": 0
     },
     "vk-bootstrap": {
-      "baseline": "0.5",
-      "port-version": 1
+      "baseline": "0.7",
+      "port-version": 0
```
Comment (FrankXie05) [inline versions/baseline.json:8357]:
```suggestion
      "baseline": "0.7",
      "port-version": 0
```
It should be added by command `./vcpkg x-add-version vk-booststrap`.

**Review by FrankXie05 — CHANGES_REQUESTED** [review]

**Comment by FrankXie05** [comment]
It's ok, I will help you finish it. :)

**Comment by FrankXie05** [comment]
Note: I will be converting your PR to draft status. When you respond, please revert to "ready for review". 
          That way, I can be aware that you've responded since you can't modify the tags.

**Review by FrankXie05 — APPROVED** [review]

**Comment by vicroms** [comment]
The `git-tree` is not a commit, it is the hash value that Git calculates for the port directory. 
For example, if you run, `git rev-parse HEAD:ports/vk-bootstrap`.

We have a vcpkg command that calculates the port's git-tree value and updates the versions database (provided that you've commited your changes) in `vcpkg x-add-version <portname>`.

**Pull request merged**